apps/shared/avifpng.c

```diff
@@ -60,7 +35,7 @@ static avifBool avifHexStringToBytes(const char * hexString, size_t hexStringLen
         }
         if (!isxdigit(hexString[i]) || !isxdigit(hexString[i + 1])) {
             avifRWDataFree(bytes);
-            fprintf(stderr, "Exif extraction failed: invalid character at " AVIF_FMT_ZU "\n", i);
+            fprintf(stderr, "Metadata extraction failed: invalid character at " AVIF_FMT_ZU "\n", i);
```

Comment:
Note: This is just a comment, not a request for change.

The code in the apps/ directory should only use the public API of the libavif library. Strictly speaking, the way this function (and related code such as `avifRemoveHeader()`) directly sets `avif->exif`, `avif->xmp` violates API encapsulation. Since C doesn't have language features for encapsulation, we can only achieve encapsulation by convention. The intention of the libavif API is that `avif->exif` and `avif->xmp` should be set by using the setter functions `avifImageSetMetadataExif()` and `avifImageSetMetadataXMP()`:

```
    // Metadata - set with avifImageSetMetadata*() before write, check .size>0 for existence after read
    avifRWData exif;
    avifRWData xmp;
```

So ideally we should not bypass the setter functions. The reason this code could be potentially problematic is that this code knows how the setter functions currently do their job and sets `avif->exif` and `avif->xmp` in the same way. If we ever change the implementation of the setter functions and the corresponding cleanup code in `avifImageDestroy()`, then this code may break `avifImageDestroy()`.

I allowed this because this avoids copying and the implementation of the setter functions is unlikely to change. (With this kind of code outside the library, we won't be able to change the implementation of the setter functions.) But we should use the setter functions when possible.